src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorPlugin.cpp

```diff
@@ -0,0 +1,160 @@
+#include "RSUHealthMonitorPlugin.h"
+
+using namespace RSUHealthMonitor;
+using namespace tmx::utils;
+
+namespace RSUHealthMonitor
+{
+
+    RSUHealthMonitorPlugin::RSUHealthMonitorPlugin(std::string name) : PluginClient(name)
+    {
+        _rsuWorker = std::make_shared<RSUHealthMonitorWorker>();
+        UpdateConfigSettings();
+        // Send SNMP call to RSU periodically at configurable interval.
+        std::thread rsuStatus_t(&RSUHealthMonitorPlugin::PeriodicRSUStatusReq, this);
+        rsuStatus_t.join();
+    }
+
+    void RSUHealthMonitorPlugin::UpdateConfigSettings()
+    {
+        PLOG(logINFO) << "Updating configuration settings.";
+
+        lock_guard<mutex> lock(_configMutex);
+        GetConfigValue<uint16_t>("Interval", _interval);
+        GetConfigValue<string>("RSUIp", _rsuIp);
+        GetConfigValue<uint16_t>("SNMPPort", _snmpPort);
+        GetConfigValue<string>("AuthPassPhrase", _authPassPhrase);
+        GetConfigValue<string>("SecurityUser", _securityUser);
+        GetConfigValue<string>("SecurityLevel", _securityLevel);
+        GetConfigValue<string>("RSUMIBVersion", _rsuMIBVersionStr);
+        boost::trim_left(_rsuMIBVersionStr);
+        boost::trim_right(_rsuMIBVersionStr);
+        // Support RSU MIB version 4.1
+        if (boost::iequals(_rsuMIBVersionStr, RSU4_1_str))
+        {
+            _rsuMibVersion = RSUMIB_V_4_1;
+        }
+        else
+        {
+            _rsuMibVersion = UNKOWN_MIB_V;
+            PLOG(logERROR) << "Uknown RSU MIB version: " <<_rsuMIBVersionStr;
+        }
+    }
+
+    void RSUHealthMonitorPlugin::OnConfigChanged(const char *key, const char *value)
+    {
+        PluginClient::OnConfigChanged(key, value);
+        UpdateConfigSettings();
+    }
+
+    void RSUHealthMonitorPlugin::PeriodicRSUStatusReq()
+    {
+        while (true)
```

Comment:
Not a fan of non exitable loops. Consider using one of the built in tmx wrappers on thread instead of thread directly like `ThreadTimer` or `ThreadWorker`. Even if you continue to use thread directly there should probably be some sort of exit condition. If you extend thread working, you can implement the DoWork function and it makes this logic unit testable since it will no longer be inside you plugin class.

Reply:
Is there any issue with retrieving the RSU info and broadcast immediately? I am trying to avoid the complexity of two threads accessing the same global variable where the RSU status is stored.

Reply:
Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/564/commits/7a4238acb369a0a8baa26b437c9b0ab59e48960a